NXDOC-2910 Release Notes for WebUI 2025.12.0 (#2620)
nuxeo/doc.nuxeo.com-platform-spaces: src/nxdoc/web-ui/web-ui-release-notes/web-ui-release-notes-2025-12-0.md

---
title: Version 2025.11.0
description: Discover what's new in Web UI 2025.12.0.
review:
  comment: ''
  date: '2026-02-11'
  status: ok
toc: true
labels:
tree_item_index: 989
hidden: true
---

{{{multiexcerpt 'matching-notes' page='web-ui-release-notes'}}}

{{! multiexcerpt name='web-ui-updates'}}

## What’s New in Web UI for LTS 2025 (Version 2025.12.0)

**Bug Fixes & Enhancements** 

- ***Normal reload for custom messages and templates:***
  - Internationalization (i18n) resources under /ui/i18n/ now bypass caching, ensuring the latest translations are always displayed on normal reload and doesnt require hard reload.

>   - Updated Service Worker behavior to prevent caching of Polymer HTML imports which ensures newly added pages and custom slots to load correctly after browser reloads.

Reviewer: ```suggestion
  - Updated Service Worker behavior to prevent caching of Polymer HTML imports, which ensures newly added pages and custom slots to load correctly after browser reloads.
```

- ***Configurable Confirmation Dialogs for Standard Actions:***

>   - Delete blob functionality now supports configurable confirmation messages which can be enabled/disabled by admins.

Reviewer: ```suggestion
  - Delete blob functionality now supports configurable confirmation messages, which can be enabled/disabled by admins.
```

- ***Automatic Production Catalog Generation:***
  - Releasing new WebUI versions now triggers automated production catalog generation, streamlining release workflows and reducing manual steps.

- ***Saved Search Optimization:***
  - Fixed an issue where the JSON payload used during save/rename operations was incorrectly modified as an array of strings.
  - Saved search operations now use the correct payload format and no longer fail silently.

>   - Long saved search names were previously truncated in the Filters dropdown.

Reviewer: ```suggestion
  - Fixed an issue where the long saved search names were truncated in the Filters dropdown.
```

  - The full saved search title is now displayed, improving readability and usability.

- ***Editing Multi‑Valued Numeric Properties:***

>   - Editing multi‑valued Number fields through the WebUI Edit layout previously caused save failures due to incorrect type handling.

Reviewer: ```suggestion
  - Resolved an issue where editing multi‑valued Number fields through the WebUI Edit layout caused save failures due to incorrect type handling.
```

>   - Values are now correctly treated as numeric types, preventing ArrayStoreException errors and ensuring documents save successfully.

Reviewer: ```suggestion
  - Values are now correctly treated as numeric types, preventing ArrayStoreException errors and ensuring documents are saved successfully.
```

---
title: Version 2025.11.0
description: Discover what's new in Web UI 2025.12.0.
review:
  comment: ''
  date: '2026-02-11'
  status: ok
toc: true
labels:
tree_item_index: 989
hidden: true
---

{{{multiexcerpt 'matching-notes' page='web-ui-release-notes'}}}

{{! multiexcerpt name='web-ui-updates'}}

## What’s New in Web UI for LTS 2025 (Version 2025.12.0)

**Bug Fixes & Enhancements** 

> - ***Normalreload for i18n Translation Labels:***

Reviewer 2: space between normal and reload

> - ***Automatic Refresh for i18n Translation Labels:***

Reviewer 2: It is not automatic refresh, but this now doesnt require a hard reload of the browser. Normal reloads will work.

>   - Internationalization (i18n) resources under /ui/i18n/ now bypass caching, ensuring the latest translations are always displayed.

Reviewer 2: latest translations are always displayed on normal reload and doesnt require hard reload

  - Users no longer need to perform a hard browser reload to see updated labels.

- ***Configurable Confirmation Dialogs for Standard Actions:***

>   - All standard Nuxeo WebUI buttons now support configurable confirmation dialogs.

Reviewer 2: Since we have only one button with this feature, its better to have that mentioned only, and not mention all buttons. Its only restricted to delete blolb button

  - Admins can enable/disable dialogs and define custom confirmation messages.
  - Behavior previously limited to “Delete” now extends to lifecycle actions and document actions, improving consistency and user safety.

- ***Automatic Production Catalog Generation:***
  - Releasing new WebUI versions now triggers automated production catalog generation, streamlining release workflows and reducing manual steps.

- ***Correct Handling of JSON Payload for Saved Searches:***
  - Fixed an issue where the JSON payload used during save/rename operations was incorrectly modified as an array of strings.
  - Saved search operations now use the correct payload format and no longer fail silently.

> - ***Saved Search Title Truncation:***

Reviewer 2: The two saved search changes can be mentioned in a single heading

  - Long saved search names were previously truncated in the Filters dropdown.
  - The full saved search title is now displayed, improving readability and usability.

- ***Editing Multi‑Valued Numeric Properties:***
  - Editing multi‑valued Number fields through the WebUI Edit layout previously caused save failures due to incorrect type handling.
  - Values are now correctly treated as numeric types, preventing ArrayStoreException errors and ensuring documents save successfully.

**Security Improvements** 

- ***Client‑Side Redirect/Phishing Mitigation:***
  - Strengthened URL validation for file downloads.

>   - Only trusted blob: and absolute http(s) URLs are allowed.

Reviewer 2: Rather than mentioning these points, we can mention there was a client-side redirect issue detected in pdfjs being used which is now fixed in the application

  - Redirects built from untrusted DOM input are now blocked to prevent phishing‑style attacks.

**Other features/Improvements**

- ***Service Worker Caching Reliability:***

>   - Updated Service Worker behavior to prevent caching of Polymer HTML imports.

Reviewer 2: This is another caching issue resolved, which does not require hard reload. So maybe this can be clubbed with the first one in a single heading?
